clicache/integration-test2/SNITests.cs

```diff
@@ -0,0 +1,127 @@
+            var clientTruststore = Config.SslClientKeyPath + @"/truststore_sni.pem";
+
+
+
+            var cacheFactory = new CacheFactory();
+            cacheFactory.Set("log-level", "none");
+            cacheFactory.Set("log-file", "c:/temp/SNITest-csharp.log");
+            cacheFactory.Set("ssl-enabled", "true");
+            cacheFactory.Set("ssl-truststore", clientTruststore);
+
+            cache_ = cacheFactory.Create();
+
+            var dc = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker-compose.exe", "-f " + Config.SniConfigPath + "/docker-compose.yml" + " up -d");
+            dc.WaitForExit();
+
+            var d = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker.exe", "exec -t geode gfsh run --file=/geode/scripts/geode-starter.gfsh");
+            d.WaitForExit();
+        }
+
+        public void Dispose()
+        {
+
+            var dockerComposeProc = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker-compose.exe", "stop");
+            dockerComposeProc.WaitForExit();
+
+            var dockerProc = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker.exe", "container prune -f");
+            dockerProc.WaitForExit();
+
+        }
+
+        private string RunDockerCommand(string dockerCommand)
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo();
+            startInfo.RedirectStandardOutput = true;
+            startInfo.UseShellExecute = false;
+            startInfo.FileName = @"C:\Program Files\Docker\Docker\resources\bin\docker.exe";
+            startInfo.Arguments = dockerCommand;
+            dockerProcess = Process.Start(startInfo);
+            String rVal = dockerProcess.StandardOutput.ReadToEnd();
+            dockerProcess.WaitForExit();
+            return rVal;
+        }
+
+        private int ParseProxyPort(string proxyString)
+        {
+            int colonPosition = proxyString.IndexOf(":");
+            string portNumberString = proxyString.Substring(colonPosition + 1);
+            return Int32.Parse(portNumberString);
+        }
+
+        [Fact (Skip = "Docker nut supported in Windows CI")]
+        public void ConnectViaProxy()
+        {
+            var portString = RunDockerCommand("port haproxy");
+            var portNumber = ParseProxyPort(portString);
+
+            cache_.GetPoolManager()
+                .CreateFactory()
+                .SetSniProxy("localhost", portNumber)
+                .AddLocator("locator-maeve", 10334)
+                .Create("pool");
+
+            var region = cache_.CreateRegionFactory(RegionShortcut.PROXY)
+                              .SetPoolName("pool")
+                              .Create<string, string>("jellyfish");
+
+            region.Put("1", "one");
+            var value = region.Get("1");
+
+            Assert.Equal("one", value);
+            cache_.Close();
+        }
+
+        [Fact (Skip = "Docker nut supported in Windows CI")]
```

Comment:
Perhaps you mean "not" rather than "nut". Also that statement is false, Docker is fully supported in Windows so why are we not skipping it?

Reply:
its Docker on GCP that is specifically the issue, but thanks for catching the typo

Reply:
I am running Docker on Windows in GCP just fine. Can I help?